os/fs/smartfs/smartfs_procfs.c

```diff
@@ -928,6 +937,57 @@ static size_t smartfs_erasemap_read(FAR struct file *filep, FAR char *buffer, si
 }
 #endif
 
+/****************************************************************************
+ * Name: smartfs_dump_all_sector
+ *
+ * Description: Performs the dump of all sector data
+ *
+ ****************************************************************************/
+#ifdef CONFIG_DEBUG_FS
+static size_t smartfs_dump_all_sector(FAR struct file *filep, FAR char *buffer, size_t buflen)
+{
+	struct mtd_smart_debug_data_s debug_data;
+	FAR struct smartfs_file_s *priv;
+
+	priv = (FAR struct smartfs_file_s *)filep->f_priv;
+
+	/* Populate the debug_data structure */
+
+	debug_data.debugcmd = SMART_DEBUG_CMD_DUMP_SECTOR;
+	debug_data.debugdata = (uint32_t)SMART_DEBUG_DUMP_ALL;
+
+	priv->level1.mount->fs_blkdriver->u.i_bops->ioctl(priv->level1.mount->fs_blkdriver, BIOC_DEBUGCMD, (unsigned long)&debug_data);
```

Comment:
It is so complicated.
why don't you change it like below
driver = priv->level1.mount->fs_blkdriver;
driver ->u.i_bops->ioctl(driver, BIOC_DEBUGCMD, (unsigned long)&debug_data);

Reply:
yes I agree but this is common style of  inode operation, many place using same style.
So it can'be allied in this PR, I will think about it later.
Thanks for your comment.

Reply:
ok. thank you